TypeSystem/flow-analysis/reachability_A12_t03.dart

```diff
@@ -22,13 +22,12 @@ class C {
 
 test(FutureOr<C> foc) async {
   if (foc is Future<C>) {
```

Comment:
I think we're checking the entire rule about `E1 is S`, in the particular case where `factor(T, S)` gets a certain kind of input. This means that it makes sense to verify the treatment of the branch that relies on `factor` as well as the other branch.

As before, the safest test would probably be `expectStaticType` expecting `Future<C>` in the "then" branch.

Reply:
Alternatively, maybe these tests are exclusively about `factor`, and we just want the bare minimum that reveals the behavior of `factor`? In that case a couple of other tests could have one empty branch, and some non-trivial code in the branch that uses `factor`.

Reply:
Thanks!